sdv/tabular/base.py

```diff
@@ -320,75 +319,69 @@ def _sample_batch(self, num_rows=None, max_retries=100, max_rows_multiplier=10,
-            total_sampled += num_to_sample
 
-            LOGGER.info('%s valid rows remaining. Resampling %s rows', remaining, num_to_sample)
+            LOGGER.info(f'{remaining} valid rows remaining. Resampling {batch_size_per_try} rows')
             sampled, num_valid = self._sample_rows(
-                num_to_sample, conditions, transformed_conditions, float_rtol, sampled
+                batch_size_per_try, conditions, transformed_conditions, float_rtol, sampled,
             )
 
             counter += 1
 
         return sampled.head(min(len(sampled), num_rows))
 
-    def _make_conditions_df(self, conditions, num_rows):
-        """Transform `conditions` into a dataframe.
+    def _make_condition_dfs(self, conditions):
+        """Transform `conditions` into a list of dataframes.
 
         Args:
-            conditions (pd.DataFrame, dict or pd.Series):
-                If this is a dictionary/Series which maps column names to the column
-                value, then this method generates `num_rows` samples, all of
-                which are conditioned on the given variables. If this is a DataFrame,
-                then it generates an output DataFrame such that each row in the output
-                is sampled conditional on the corresponding row in the input.
-            num_rows (int):
-                Number of rows to sample. If a conditions dataframe is given, this must
-                either be ``None`` or match the length of the ``conditions`` dataframe.
+            conditions (list[sdv.sampling.Condition]):
+                A list of `sdv.sampling.Condition`, where each `Condition` object
+                represents a desired column value mapping and the number of rows
+                to generate for that condition.
 
         Returns:
-            pandas.DataFrame:
-                `conditions` as a dataframe.
+            list[pandas.DataFrame]:
+                A list of `conditions` as dataframes.
         """
-        n_rows = num_rows if num_rows else self._num_rows
-        if isinstance(conditions, pd.Series):
-            conditions = pd.DataFrame([conditions] * n_rows)
-
-        elif isinstance(conditions, dict):
-            conditions = pd.DataFrame(conditions, index=range(n_rows))
-
-        elif not isinstance(conditions, pd.DataFrame):
-            raise TypeError('`conditions` must be a dataframe, a dictionary or a pandas series.')
-
-        elif num_rows is not None and len(conditions) != num_rows:
-            raise ValueError(
-                'If `conditions` is a `DataFrame`, `num_rows` must be `None` or match its lenght.')
+        condition_dataframes = defaultdict(list)
+        for condition in conditions:
+            column_values = condition.get_column_values()
+            condition_dataframes[tuple(column_values.keys())].append(
+                pd.DataFrame(
+                    condition.get_column_values(),
+                    index=range(condition.get_num_rows())
```

Comment:
Doesn't this get set automatically? Do we really need this line?

Reply:
A condition could be defined as
```
condition = Condition({'a': 1}, num_rows=5)
```
which means that the user wants 5 rows with the condition that column `a` is 1, so in the dataframe we want `{'a': 1}` to repeat 5 times. Setting the index to range(5) does that.